addons/stock_account/tests/test_anglo_saxon_valuation_reconciliation.py

```diff
@@ -0,0 +1,116 @@
+
+        products = [self.test_product_delivery, self.test_product_delivery_2]
+
+        # Create invoices
+        invoices = []
+        for product in products:
+            move_form = Form(self.env["account.move"].with_context(default_move_type="out_invoice"))
+            move_form.partner_id = self.partner_a
+            move_form.currency_id = self.currency_data["currency"]
+            with move_form.invoice_line_ids.new() as line_form:
+                line_form.product_id = product
+            invoices.append(move_form.save())
+
+        # Creating the aml that would be reconciled twice if no reconciliation checking step is implemented
+        broken_aml = self.env['account.move.line'].create({
+            'name': "broken aml",
+            'product_id': self.test_product_delivery.id,
+            'account_id': self.stock_account_product_categ["property_stock_account_output_categ_id"].id,
+            'move_id': invoices[0].id
+        })
+
+        # Creating dummy stock.move to bypass other check
+        stock_location = self.env['stock.warehouse'].search([
+            ('company_id', '=', self.env.company.id),
+        ], limit=1).lot_stock_id
+
+        product = products[0]
+
+        out_picking = self.env['stock.picking'].create({
+            'location_id': stock_location.id,
+            'location_dest_id': self.ref('stock.stock_location_customers'),
+            'picking_type_id': stock_location.warehouse_id.out_type_id.id,
+        })
+
+        sm = self.env['stock.move'].create({
+            'name': product.name,
+            'product_id': product.id,
+            'product_uom_qty': 1,
+            'product_uom': product.uom_id.id,
+            'location_id': out_picking.location_id.id,
+            'location_dest_id': out_picking.location_dest_id.id,
+            'picking_id': out_picking.id,
+        })
+
+        # Patched methods to bypass other checks
+        def _stock_account_get_last_step_stock_moves(self):
+            return sm
+
+        def _reconcile_plan(self, reconciliation_plan):
+            for amls in reconciliation_plan:
+                for aml in amls:
+                    if aml.reconciled:
+                        raise AssertionError("Journal items should not be reconciled twice")
+                    aml.reconciled = True
+
+        def _get_all_related_sm(self, prod):
+            return self
+
+        def _get_all_related_aml(self):
+            return broken_aml
+
+        # Running _post with patched methods (which calls _stock_account_anglo_saxon_reconcile_valuation)
+        with (
+            patch.object(AccountMoveLine, '_reconcile_plan', _reconcile_plan),
+            patch.object(AccountMove, '_stock_account_get_last_step_stock_moves', _stock_account_get_last_step_stock_moves),
+            patch.object(StockMove, '_get_all_related_sm', _get_all_related_sm),
+            patch.object(StockMove, '_get_all_related_aml', _get_all_related_aml),
+        ):
```

Comment:
This is not a good way to test a flow. The goal is to reproduce a bug by calling actual business methods. Mocking methods do not prove the scenario is robust after your patch. The ticket mention an invoice impossible to confirm. I would:
* prepare some data (you did it already) 
* create an invoice
* confirm it 
* check the state of the invoice is correct